test/hotspot/jtreg/runtime/cds/appcds/aotClassLinking/StringConcatStress.java

```diff
@@ -0,0 +1,3162 @@
+/*
+ * Copyright (c) 2024, 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+/*
+ * @test id=static
+ * @requires vm.cds.write.archived.java.heap
+ * @library /test/lib
+ * @build StringConcatStress
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller -jar app.jar StringConcatStressApp
+ * @run driver StringConcatStress STATIC
+ */
+
+/*
+ * @test id=aot
+ * @requires vm.cds
+ * @requires vm.cds.supports.aot.class.linking
```

Comment:
From the VMProps.java in the mainline:
```
    protected String vmCDSSupportsAOTClassLinking() {
      // Currently, the VM supports AOTClassLinking as long as it's able to write archived java heap.
      return vmCDSCanWriteArchivedJavaHeap();
    }
```
The `vmCDSCanWriteArchivedJavaHeap()` already checks for `vmCDS()`.
So it seems the `@requires vm.cds` is redundant.